lib/Dialect/TTIR/Transforms/Passes.cpp

```diff
@@ -1449,4 +1516,117 @@ class TTIRLoadSystemDesc
   }
 };
 
+class TTIRFlattenZYReductionRewriter : public OpRewritePattern<MeanOp> {
+public:
+  using OpRewritePattern<MeanOp>::OpRewritePattern;
+
+  ReshapeOp createReshapeOp(PatternRewriter &rewriter, Location loc,
+                            Value input, ::llvm::ArrayRef<int64_t> shapei64,
+                            ::mlir::ArrayAttr operandConstraints) const {
+    auto ty = mlir::cast<RankedTensorType>(input.getType());
+    auto output =
+        rewriter.create<tensor::EmptyOp>(loc, shapei64, ty.getElementType());
+
+    auto shape_attr = rewriter.getI32ArrayAttr(
+        {static_cast<int32_t>(shapei64[0]), static_cast<int32_t>(shapei64[1]),
+         static_cast<int32_t>(shapei64[2]), static_cast<int32_t>(shapei64[3])});
+    return rewriter.create<ttir::ReshapeOp>(
+        loc, output.getType(), input, output, shape_attr, operandConstraints);
+  }
+
+  LogicalResult matchAndRewrite(MeanOp reduce1,
+                                PatternRewriter &rewriter) const final {
+
+    if (!reduce1.getDimArg().has_value()) {
+      return failure();
+    }
+
+    if (reduce1.getDimArg().value().size() != 1) {
+      return failure();
+    }
+
+    int64_t reduce1_dim =
+        mlir::cast<IntegerAttr>(reduce1.getDimArg().value().getValue()[0])
+            .getInt();
+    if (reduce1_dim != -3) {
+      return failure();
+    }
+
+    MeanOp reduce2;
+    bool found = false;
+    for (auto user : reduce1->getUsers()) {
+      user->getName().dump();
+      if (isa<MeanOp>(user)) {
+        reduce2 = mlir::cast<MeanOp>(user);
+        found = true;
+      }
+    }
```

Comment:
What if there are multiple users? We probably need to count the number of users and bail if > 1